designs/credentials/credentials-management.md

```diff
@@ -0,0 +1,65 @@
+# Credentials Management
+
+This outlines how the Toolkit produces and obtains credentials.
+
+## Terminology
+
+### Credentials
+
+Credentials allow the Toolkit to interact with AWS on a user's behalf. Service clients require a [Credentials](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Credentials.html) object, and the Toolkit obtains these objects through Credentials Providers.
+
+### Credentials Provider
+
+Credentials Providers are how the toolkit abstracts away different ways of obtaining a user's credentials, and produces Credentials objects. For example, a Shared Credentials Provider knows how to obtain credentials for a specific profile within [Shared Credentials Files](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/loading-node-credentials-shared.html).
+
+A Credentials Provider is produced by a Credentials Provider Factory
+
+### Credentials Provider Factory
+
+A factory is capable of producing one or more Credentials Providers for a single credentials type. For example, a Shared Credentials related factory produces Credentials Providers for each profile found within shared credentials files.
+
+### Credentials Provider Id
+
+All Credentials Providers are uniquely identified by a Credentials Provider Id. These take the format `(credentials type):(credentials type id)`. For example, Shared Credentials Providers have the type `profile`, and the id represents the profile name. So if shared credentials contained a profile named "foo", its corresponding Credentials Provider Id would be `profile:foo`.
```

Comment:
Should we find a different separator? Colons are allowed in profile names.

Reply:
I couldn't find a spec for profile names. Do you have a link?

Reply:
Can't find one either, found this through trial and error. Was going to propose a colon and a space but I just tested it and spaces work :/

Reply:
Discussed offline, we'll switch this to a pipe `|`